internal/entitlement/postgresadapter/entitlement.go

```diff
@@ -26,7 +28,12 @@ type entitlementDBAdapter struct {
 	db *db.Client
 }
 
-func NewPostgresEntitlementRepo(db *db.Client) entitlement.EntitlementRepo {
+type repo interface {
```

Comment:
maybe not, but I would prefer to not have this interface here depending on entitlement + balanceworker.

Instead, we should not have interface constraints here, and whenever we want to use the return value of NewPostgresEntitlementRepo the checking should happen there.

That would hugely decrease the dependency hell we have between the packages.

This interface constraining also necessitates the splitting of registry/builder just because we are using the repo directly inside the balanceworker.

Reply:
I think registry & builder being combined was the issue actually. If you think about the direction of dependencies, implementations depend on abstractions so it's only right that we depend here on the interfaces. The issue was that registry depended in the both & then it was used in packages defining further interfaces we wanted to implement. 

I'm ok with using two type asserts instead of `repo` but IMO that's a technicality, but i'm confident this is the correct direction of dependencies.

Reply:
We can talk about this IRL, but I don't think that this interface should be mandated here. Right now this is how it is, I was just signaling that this is not something I am confident is a good approach.